### PR TITLE
Make this compile with swift 2.1(and 2.2), apply some suggestions from analyzer(mainly var -> let)

### PR DESCRIPTION
**Pull request by TimurBK**
I've made some adjustments to be able to compile this project in Xcode 7.2(and 7.3). 
Also unless I messed something up during this, Swift is actually faster then C if safety checks are disabled.

Here are some results(rMBP 2.7 GHz i7, 16 Gb, OS X 10.10.5, Xcode 7.2.1):
```
-O + safety checks disabled for Swift, -Ofast for C

Final Results:
--------------
Swift sort: 0.006770 s (± 0.001038)
Quick sort: 0.008186 s (± 0.001261)
Heap sort: 0.009233 s (± 0.000980)
Insertion sort: 1.842363 s (± 0.023623)
Selection sort: 4.156963 s (± 0.036717)

Final Results:
--------------
Swift sort: 0.006470 s (± 0.000856)
Quick sort: 0.007506 s (± 0.000880)
Heap sort: 0.009371 s (± 0.001162)
Insertion sort: 1.838529 s (± 0.010603)
Selection sort: 4.161174 s (± 0.021795)

Final Results:
--------------
C sort: 0.013883 s (± 0.001380)
Quick sort: 0.007780 s (± 0.001132)
Heap sort: 0.010620 s (± 0.001426)
Insertion sort: 5.612749 s (± 0.023890)
Selection sort: 3.475682 s (± 0.013629)


Final Results:
--------------
C sort: 0.014077 s (± 0.001787)
Quick sort: 0.008084 s (± 0.001297)
Heap sort: 0.010950 s (± 0.001498)
Insertion sort: 5.598251 s (± 0.021549)
Selection sort: 3.471042 s (± 0.010570)
```
```
-O3

Final Results:
--------------
C sort: 0.013798 s (± 0.001547)
Quick sort: 0.007956 s (± 0.001220)
Heap sort: 0.011105 s (± 0.001600)
Insertion sort: 5.601278 s (± 0.028258)
Selection sort: 3.472231 s (± 0.008252)

Final Results:
--------------
C sort: 0.013337 s (± 0.001404)
Quick sort: 0.008342 s (± 0.001442)
Heap sort: 0.011014 s (± 0.001489)
Insertion sort: 5.613515 s (± 0.059526)
Selection sort: 3.475649 s (± 0.020319)


Final Results:
--------------
Swift sort: 0.006768 s (± 0.000990)
Quick sort: 0.009044 s (± 0.001269)
Heap sort: 0.012063 s (± 0.001598)
Insertion sort: 3.453265 s (± 0.017760)
Selection sort: 4.146754 s (± 0.009792)

Final Results:
--------------
Swift sort: 0.006437 s (± 0.000823)
Quick sort: 0.008453 s (± 0.000799)
Heap sort: 0.011670 s (± 0.001525)
Insertion sort: 3.451959 s (± 0.012546)
Selection sort: 4.157997 s (± 0.015493)
```

### CODE REVIEW

**Comment by TimurBK** [comment]
Strange that Travis fails to compile sources.

**Comment by jessesquires** [comment]
Thanks @TimurBK ! :smile: 

I think travis failed because the [`.travis.yml`](https://github.com/jessesquires/swift-sorts/blob/master/.travis.yml) was out-dated. I've just updated it.

Can you rebase on `master` and see if tests pass now?

**Comment by TimurBK** [comment]
Tests are green now. :)

**Comment by jessesquires** [comment]
>Tests are green now. :)

:tada: woo!

>One thing that I left untouched is var parameters. 

`var` params will eventually be removed from Swift. let's remove the `var` declarations. (which will default to `let` — no need to put `let` explicitly)

File: SwiftSortsTests/SwiftSortsTests.swift [hunk 1]
```diff
@@ -26,42 +26,42 @@ class SwiftSortsTests: XCTestCase {
     }
     
     func testSwiftSort() {
-        XCTAssertFalse(equal(self.sortedArray, self.unsortedArray!), "Arrays should not be equal before sorting")
+        XCTAssertFalse(self.sortedArray == self.unsortedArray!, "Arrays should not be equal before sorting")
```
Comment (jessesquires) [inline SwiftSortsTests/SwiftSortsTests.swift]: each of these can become:

```swift
XCTAssertEqual(self.sortedArray, self.unsortedArray!, "...")
```

**Comment by TimurBK** [comment]
Updated. Replacing var parameters with let and then creating var inside function slowed them down a bit, but not so much. 
Also it seems something is wrong in my swift's selection sort results, they should be around 3.3 sec. I'll rerun benchmarks a bit later, probably tomorrow.

**Comment by jessesquires** [comment]
Thanks so much for working on this @TimurBK ! :smile: 

I'm going to go ahead and merge this. But, please feel free send another PR or comment here with your benchmark results. :+1:

**Comment by TimurBK** [comment]
It took a bit longer, but here are my results. :D
My machine is Retina MBP Early 2013 (2.7 GHz/16 GB/512 GB SSD). OS X 10.11.4, Xcode 7.3(7D175).

Swift with enabled safety checks.
```
Final Results:
--------------
Swift sort: 0.006884 s (± 0.000963)
Quick sort: 0.009267 s (± 0.001376)
Heap sort: 0.010616 s (± 0.000383)
Insertion sort: 3.450161 s (± 0.036773)
Selection sort: 4.141659 s (± 0.028353)
```
Swift with disabled safety checks
```
Final Results:
--------------
Swift sort: 0.006889 s (± 0.000862)
Quick sort: 0.007762 s (± 0.001107)
Heap sort: 0.009638 s (± 0.001420)
Insertion sort: 2.080200 s (± 0.022648)
Selection sort: 2.925644 s (± 0.006636)
```

C -O3
```
Final Results:
--------------
C sort: 0.014441 s (± 0.001277)
Quick sort: 0.008361 s (± 0.001419)
Heap sort: 0.011258 s (± 0.001558)
Insertion sort: 5.561654 s (± 0.087450)
Selection sort: 3.134947 s (± 0.057520)
```
C, aggressive optimizations(doesn't seem to make much difference)
```
Final Results:
--------------
C sort: 0.014336 s (± 0.000937)
Quick sort: 0.007999 s (± 0.001345)
Heap sort: 0.011606 s (± 0.001717)
Insertion sort: 5.524653 s (± 0.023527)
Selection sort: 3.112783 s (± 0.009751)
```

**Comment by jessesquires** [comment]
@TimurBK :tada: :+1:

**Pull request merged**